homework-submission/my-app/src/App.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import './App.css';
+import TodoItems from './Components/createDynamicItems';
+import Button from './Components/button';
+
+class Todolist extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [
+        {
+          name: 'REACT Lesson',
+          deadline: 23 - 3 - 2019,
+          key: Date.now(),
+        },
+        {
+          name: 'Wach movie',
+          deadline: 23 - 3 - 2019,
+          key: Date.now() + 1,
+        },
+        {
+          name: 'Football',
+          deadline: 23 - 3 - 2019,
+          key: Date.now() + 2,
+        },
+      ],
+    };
+    this.addItem = this.addItem.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+  }
+
+  addItem(e) {
+    if (this._inputElement.value !== '') {
+      let newItem = {
+        name: this._inputElement.value,
+        key: Date.now(),
+      };
+      this.setState(PrevState => {
+        return {
+          items: PrevState.items.concat(newItem),
+        };
+      });
+    }
+    this._inputElement.value = ' ';
+    e.preventDefault();
+  }
+
+  deleteItem(key) {
+    const filteredItems = this.state.items.filter(item => {
+      return item.key !== key;
```

Comment:
you may see some of the confusion here.
people used to use the `id` as the unique identity for any array contains number of objects, and in this case any other developer working on your code won't guess he have to use the `key` as a uniquely identifying property

So in a bigger program _(which you will do)_ this may distract you from other things as well _(you may use key for another purpose)_ and you will have an `id` from the server.